src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -306,69 +386,120 @@ class MoneyRequestAmountPage extends React.Component {
             .value();
     }
 
+    navigateBack() {
+        Navigation.goBack(this.isEditing ? ROUTES.getMoneyRequestConfirmationRoute(this.iouType, this.reportID) : null);
+    }
+
     navigateToCurrencySelectionPage() {
         // Remove query from the route and encode it.
         const activeRoute = encodeURIComponent(Navigation.getActiveRoute().replace(/\?.*/, ''));
-        if (this.props.hasMultipleParticipants) {
-            return Navigation.navigate(ROUTES.getIouBillCurrencyRoute(this.props.reportID, this.state.selectedCurrencyCode, activeRoute));
+        Navigation.navigate(ROUTES.getMoneyRequestCurrencyRoute(this.iouType, this.reportID, this.state.selectedCurrencyCode, activeRoute));
+    }
+
+    navigateToNextPage() {
+        const amountInSmallestCurrencyUnits = CurrencyUtils.convertToSmallestUnit(this.state.selectedCurrencyCode, Number.parseFloat(this.state.amount));
+        IOU.setMoneyRequestAmount(amountInSmallestCurrencyUnits);
+        IOU.setMoneyRequestCurrency(this.state.selectedCurrencyCode);
+
+        if (this.isEditing) {
+            Navigation.goBack(ROUTES.getMoneyRequestConfirmationRoute(this.iouType, this.reportID));
```

Comment:
```suggestion
            Navigation.goBack(ROUTES.getMoneyRequestConfirmationRoute(this.iouType, this.reportID), true);
```

Reply:
No need to enforce the fallback for this one

Reply:
We will go back in edit amount only, so it will go back to confirmation page, even without the enforce.

Reply:
We always want to go to the confirmation page, if it's always then we should enforce it. The confirmation page is not used as a fallback in this case.

Reply:
Sure